drracket-tool-lib/drracket/private/standalone-module-browser.rkt

```diff
@@ -1264,38 +1243,35 @@
       (define last-size #f)
       
       (define/private (name->label)
-        (let ([this-size (send pb get-name-length)])
-          (cond
-            [(eq? this-size last-size) last-name]
-            [else
-             (set! last-size this-size)
-             (set! last-name
-                   (case last-size
-                     [(short)
-                      (if (string=? word "")
-                          ""
-                          (string (string-ref word 0)))]
-                     [(medium)
-                      (let ([m (regexp-match #rx"^(.*)\\.[^.]*$" word)])
-                        (let ([short-name (if m (cadr m) word)])
-                          (if (string=? short-name "")
-                              ""
-                              (let ([ms (regexp-match* #rx"-[^-]*" short-name)])
-                                (cond
-                                  [(null? ms)
-                                   (substring short-name 0 (min 2 (string-length short-name)))]
-                                  [else
-                                   (apply string-append
-                                          (cons (substring short-name 0 1)
-                                                (map (λ (x) (substring x 1 2))
-                                                     ms)))])))))]
-                     [(long) word]
-                     [(very-long)  
-                      (string-append
-                       word
-                       ": "
-                       (format "~s" require-phases))]))
-             last-name])))
+        (define this-size (send pb get-name-length))
+        (cond
+          [(eq? this-size last-size) last-name]
+          [else
+           (set! last-size this-size)
+           (set!
+            last-name
```

Comment:
this is probably a bad place to break a `set!`

Reply:
I agree (cc @sorawee)